signal2html/models.py

```diff
@@ -87,6 +89,27 @@ class Thread:
     mms: List[MMSMessageRecord] = field(default_factory=lambda: [])
     sms: List[SMSMessageRecord] = field(default_factory=lambda: [])
 
+    @property
+    def sanephone(self):
+        """Return a sanitized phone number suitable for use as filename, and fallback on rid.
+
+        NOTE: Phone numbers can be alphanumerical characters especially coming over SMS
+        Different strings can still clash in principle if they sanitize to the same string.
+        """
+        if self.recipient.phone:
+            return sub('[^]\\w!@#$%^&\'`.=+{}~()[-]', '_', normalize('NFKC', self.recipient.phone.strip()).lstrip(".#"))
+        else:
+            return '#' + self.recipient.recipientId._id
+
     @property
     def name(self):
+        """Return the raw name or other useful identifier, suitable for display."""
         return self.recipient.name.strip()
+
+    @property
+    def sanename(self):
+        """Return a sanitized name or other useful identifier, suitable for use as filename, and fallback on rid."""
+        if self.recipient.name:
+            return sub('[^]\\w!@#$%^&\'`.=+{}~()[-]', '_', normalize('NFKC', self.recipient.name.strip()).lstrip(".#"))
+        else:
+            return '#' + self.recipient.recipientId._id
```

Comment:
Since this is essentially the same functionality in the two properties, It would probably be better to combine them in a single method. Something like:

```python
def _sanitize(self, text):
    """Sanitize string"""
    clean = normalize('NFKC', text.strip())
    clean = clean.lstrip(".#")
    clean = sub('[^]\\w!@#$%^&\'`.=+{}~()[-]', '_', clean)
    return clean
```

Reply:
Agreed